main.go

```diff
@@ -25,16 +23,18 @@ func main() {
 	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
 	flag.Parse()
 
-	opts := &plugin.ServeOpts{ProviderFunc: provider.New()}
-
 	if debugMode {
 		// TODO: update this string with the full name of your provider as used in your configs
-		err := plugin.Debug(context.Background(), "registry.terraform.io/hashicorp/hcp", opts)
-		if err != nil {
-			log.Fatal(err.Error())
-		}
+		plugin.Serve(&plugin.ServeOpts{
```

Comment:
Where's this change from? 👀

Reply:
Also could probably delete that TODO on L27

Reply:
Yea this was a fun one 😬 
```
main.go:32:10: SA1019: plugin.Debug is deprecated: Use the Serve function with the ServeOpts Debug field instead. (staticcheck)
                err := plugin.Debug(context.Background(), "registry.terraform.io/hashicorp/hcp", opts)
                       ^
```

I followed what was recommended here: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/plugin#Debug